domainlab/algos/trainers/a_trainer.py

```diff
@@ -301,6 +314,15 @@ def cal_reg_loss(self, tensor_x, tensor_y, tensor_d, others=None):
         list_loss_tensor = list_reg_loss_model_tensor + \
             list_reg_loss_trainer_tensor
         list_mu = list_mu_model + list_mu_trainer
+        # ERM return a tensor of all zeros, delete here
+        if len(list_mu) > 1:
+            list_boolean_zero = [torch.all(torch.eq(list_loss_tensor[i], 0)).item()
+                                 for i in range(len(list_mu))]
+            list_loss_tensor = [list_loss_tensor[i] for (i, flag) in
+                                enumerate(list_boolean_zero) if not flag]
+            list_mu = [list_mu[i] for (i, flag) in enumerate(list_boolean_zero) if not flag]
+        if self.dict_multiplier:
+            list_mu = list(self.dict_multiplier.values())
```

Comment:
this is bad implementation, will list(self.dict_multiplier.values) return the same list order each time? since mhof is updating the self.dict_multiplier, as long as the order each time is the same (even if the order does not agree with the order list_mu) returns. 

Alexej suggest to use OrderDictionary

Reply:
OrderedDict preserves the order in which items are inserte, so the same as dicitonary here

Reply:
list(dict.keys()) will return keys in the same order they were inserted each time it's called, as long as the dictionary hasn't been modified.